# The Vestibular System 1: Mathematical Description

* 3 DOF Translations -> Otoliths (Utricle, Saccule) 
* 3 DOF Rotations -> 3 Semicircular Canals

<img src="data/images/labyrinth_labelled.png", width=350px>

## Semicircular Canals

The three semicircular canals (SCCs) on each side of the head transduce angular velocity. To describe this transduction mathematically, each canal can be described by a vector perpendicular to the surface spanned by
the canal $\vec{n}$.

### Orientation of all 3 Canals, on the Right Side of the Head


In [2]:
# Import the required packages
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt

# Make the display easy to read
% precision 2

'%.2f'

In [3]:
# Enter the information for the canalas into a Python dictionary.
# The data are taken from Blanks RH, Curthoys IS, and Markham CH. Planar relationships of
# the semicircular canals in man. Acta Otolaryngol  80:185-196, 1975.
Canals = {'info': 'The matrix rows describe horizontal, anterior, and posterior canal orientation',
 'right': np.array(
        [[0.365,  0.158, -0.905], 
         [0.652,  0.753, -0.017],
         [0.757, -0.561,  0.320]]),
 'left': np.array(
        [[-0.365,  0.158,  0.905],
         [-0.652,  0.753,  0.017],
         [-0.757, -0.561, -0.320]])}

# Normalize these vectors (just a small correction):
for side in ['right', 'left']:
    Canals[side] = (Canals[side].T / np.sqrt(np.sum(Canals[side]**2, axis=1))).T

# Show the results for the right SCCs:
print(Canals['info'])
print(Canals['right'])

The matrix rows describe horizontal, anterior, and posterior canal orientation
[[ 0.37  0.16 -0.92]
 [ 0.65  0.76 -0.02]
 [ 0.76 -0.56  0.32]]


# Example: Stimulation of the Right Horizontal Canal, by Rotation to the Right, with 100 deg/2

\begin{equation}
    stim = \vec{\omega} \cdot \vec{n}
\end{equation}

$\vec{\omega}$ ... relativ to the *head*!

<img src="data/images/SCC_stimulation_with_omega_.jpg", width=150px>



In [9]:
omega = np.r_[0, 0, -100]
stim = omega @ Canals['right'][0]

# Before Python 3.5:
# stim = np.dot(omega, Canals['right'][0])

print('The angular velocity sensed by the right horizontal canal is {0:3.1f} deg/s.'.format(stim))

The angular velocity sensed by the right horizontal canal is 91.5 deg/s.


<font color='red'>
<img src="data/images/matlab_logo_s.jpg", width=50px, align='left'>*Matlab Equivalent Commands*:
</font>

    % Define the semicircular canals
    Canals(1).side = 'right';
    Canals(1).rows = {'horizontal canal'; 'anterior canal'; 'posterior canal'};
    Canals(1).orientation = [	.365	 .158	-.905
        .652	 .753	-.017
        .757	-.561	 .320];
    Canals(2).side = 'left';
    Canals(2).rows = {'hor'; 'ant'; 'post'};
    Canals(2).orientation = [-.365	  .158 	 .905
        -.652	  .753 	 .017
        -.757 	 -.561	-.320];

    % Normalize the canal-vectors (only a tiny correction):
    for ii = 1:2
        for jj = 1:3
            Canals(ii).orientation(jj,:) = ...
                Canals(ii).orientation(jj,:) / norm( Canals(ii).orientation(jj,:) );
        end
    end

    % Calculate the stimulation
    omega = [0, 0, -100];
    stim = omega * Canals(1).orientation(1,:).T;

## Otoliths

The description of the otoliths is more involved, since each haircell on the otolith membrane senses the local strength of the gravito-inertal force. The direction of maximum sensitivity can for each haircell be described by a *Sensitivity Vector* $\vec{n}$.

<img src="data/images/UtriShapeHair.jpg", width=600px>


\begin{equation}
    \overrightarrow{GIF} = \vec{g} + \frac{d^2\vec{x}}{dt^2}
\end{equation}

\begin{equation}
    stim \approx \overrightarrow{GIF} \cdot \vec{n}
\end{equation}


- For the utricle, the sensitivity vectors lie approx. in the horizontal plane
- For the saccule, the sensitivity vectors lie approx. in the vertical plane
